pkg/agent/variables.go

```diff
@@ -28,9 +29,28 @@ func getCustomDataVariables(config *datamodel.NodeBootstrappingConfiguration) pa
 			"reconcilePrivateHostsService": getBase64EncodedGzippedCustomScript(reconcilePrivateHostsService, config),
 			"configureAzure0Script":        getBase64EncodedGzippedCustomScript(kubernetesConfigAzure0Script, config),
 		},
+		// customData defined here is mainly used for Windows
+		"customData": paramsMap{
+			"tenantID":                    config.TenantID,
+			"subscriptionId":              config.SubscriptionID,
+			"resourceGroup":               config.ResourceGroupName,
+			"location":                    cs.Location,
+			"vmType":                      cs.Properties.GetVMType(),
+			"subnetName":                  cs.Properties.GetSubnetName(),
+			"nsgName":                     cs.Properties.GetNSGName(),
+			"virtualNetworkName":          cs.Properties.GetVirtualNetworkName(),
+			"routeTableName":              cs.Properties.GetRouteTableName(),
+			"primaryAvailabilitySetName":  cs.Properties.GetPrimaryAvailabilitySetName(),
+			"primaryScaleSetName":         cs.Properties.GetPrimaryScaleSetName(),
+			"useManagedIdentityExtension": useManagedIdentity(cs),
+			"useInstanceMetadata":         useInstanceMetadata(cs),
+			"loadBalancerSku":             cs.Properties.OrchestratorProfile.KubernetesConfig.LoadBalancerSku,
+			"excludeMasterFromStandardLB": true,
+			"enableTelemetry":             cs.Properties.FeatureFlags.IsFeatureEnabled("EnableTelemetry"),
+		},
```

Comment:
no, please get these from NodeBootstrappingConfiguration instead of keep relying on cs.properties. 
Also for linux, these are not needed for custom data generation, it's only for cse command. I'd be interested in why we need them for windows scenario

Reply:
Maybe we are still on the way to follow Linux's work. We just first copied the case from aks-engnie

Reply:
Add a comment for the usage of windows